packages/react/src/components/Button/Button.js

```diff
@@ -32,7 +34,8 @@ const Button = React.forwardRef(function Button(
 ) {
   const buttonClasses = classNames(className, {
     [`${prefix}--btn`]: true,
-    [`${prefix}--btn--sm`]: small,
+    [`${prefix}--btn--field`]: size === 'field',
+    [`${prefix}--btn--sm`]: size === 'small' || small,
```

Comment:
Could we add a test for this new logic to make sure we don't lose support for this during the lifecycle of the deprecation?